File: SPAwebsite/Models/SimpleDate.ts
class SimpleDate
{
    private y: number;
    private m: number;
    private d: number;
    get Day():number
    {
        return this.d;
    }
    get Month():number {
        return this.m;
    }
    get Year():number {
        return this.y;
    }
    set Day(dayValue: number)
    {
        if (dayValue >= 1 && dayValue <= 31)
            this.d = dayValue;
    }
    set Month(monthValue: number) {
        if (monthValue >= 1 && monthValue <= 12)
            this.m = monthValue;
    }
    set Year(yearValue: number)
    {
        if (yearValue > 2006)
            this.y = yearValue;
    }
    constructor(Day: number, Month: number, Year: number)
    {
        this.Day = Day;
        this.Month = Month;
        this.Year = Year;
    }

    get ToDate(): Date
    {
        return new Date(this.Year, this.Month, this.Day);
    }
    toString()
    {
        return this.Day + "." + this.Month + "." + this.Year;
    }
}

File: FieldReport/Models/SimpleDate.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace FieldReport.Models
{
    public class SimpleDate
    {
        public SimpleDate()
        {
        }

        public SimpleDate(DateTime date)
        {
            Day = (ushort)date.Day;
            Month = (ushort)date.Month;
            Year = (ushort)date.Year;
        }

        [Required]
        [Range(0,31)]
        public ushort Day { get; set; }

        [Required]
        [Range(0, 12)]
        public ushort Month { get; set; }

        [Required]
        [Range(2010, 2100)]
        public ushort Year { get; set; }

        public DateTime ToDateTime
        {
            get { return new DateTime(Year, Month, Day); }
        }

        public DateTime SetDate
        {
            set
            {
                this.Day = (ushort)value.Day;
                this.Year = (ushort)value.Year;
                this.Month = (ushort)value.Month;
            }
        }

        public static bool operator ==(SimpleDate sd1, SimpleDate sd2)
        {
            return sd1.Day == sd2.Day && sd1.Month == sd2.Month && sd1.Year == sd2.Year;
        }

        public static bool operator !=(SimpleDate sd1, SimpleDate sd2)
        {
            return sd1.Day != sd2.Day || sd1.Month != sd2.Month || sd1.Year != sd2.Year;
        }

        public override int GetHashCode()
        {
            return base.GetHashCode();
        }

        public override bool Equals(object obj)
        {
            string type = obj.GetType().Name;
            switch (type)
            {
                case "DateTime":
                    {
                        SimpleDate SD = ((DateTime)obj).ToSimpleDate();
                        return SD.Day == this.Day && SD.Month == this.Month && this.Year == SD.Year;
                    }

                case "SimpleDate":
                    {
                        SimpleDate SD = obj as SimpleDate;
                        return SD.Day == this.Day && SD.Month == this.Month && this.Year == SD.Year;
                    }

                default:
                    {
                        return false;
                    }
            }
        }
    }
}